### DISCUSSION TITLE
함수를 분리하는 적절한 기준은 무엇일까요?

### DISCUSSION THREAD

**Discussion by wonsss** [woowacourse-study/2022-code-review-study#11] [Lotto] [answered]
함수를 분리하는 적절한 기준은 무엇일까요?

**Comment by lokba** [comment]
최근에, `모던 자바스크립트 딥다이브`에서 메서드의 의미에 대해서 배웠는데요.
제가 기억하기론, 메서드란 상태 데이터를 참조하고 조작할 수 있는 동작(behavior)이라고 기억하고있습니다. 

상태 데이터를 참조하지도 않고, 조작하지도 않는 연관성이 없는 메서드라면 함수로 분리하는 것이 적절하지 않나 싶네요.
만약, 함수로 분리하는게 애매한 상황이라면 static 키워드를 사용하는것도 하나의 방법이 될것 같기도 하네요.

**Reply by lokba** [reply]
위 내용은 제가 내용을 잘못 이해한 답변입니다. 넘어가세요ㅠㅠ

**Comment by moonheekim0118** [comment]
저도 위의 록바의 답변과 일맥상통하는 것 같은데요, 개인적으로 이 함수를 분리했을 때 변하는 값 혹은 새롭게 생성되는 값이 있을까? 에 따라서 분리하는 것 같습니다.

**Comment by wonsss** [comment, marked as answer]
어느 리뷰어님의 의견에 따르면, 함수를 분리해야 하는 목적은 `외부에서의 접근수단 제공` 및 `여러 곳에서 호출`을 기준으로 판단해야 한다고 합니다. 단순히 기능이 다르다는 이유만으로 잘게 분리하면 오히려 혼란을 줄 수도 있다고 하네요..
